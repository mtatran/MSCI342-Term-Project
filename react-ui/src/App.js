import React, { Component } from 'react'
import LogIn from "./components/Layouts/LogIn"
import Dashboard from "./components/Dashboard/Dashboard"
import Signup from "./components/Layouts/Signup"
import './App.css'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";


class App extends Component {
  state = {
    postgres: null,
    error: null
  }
  render() {
    
    return (
      <div className="App">
        <Router>
      <Switch>
        <Route path="/signup" component={Signup}>
          {/* <Signup /> */}
        </Route>
        <Route path="/login" component={LogIn}>
          {/* <LogIn /> */}
        </Route>
        <Route exact path="/dashboard/:email/:userType/:firstName/:personId/:accessToken" component={Dashboard}>
          {/* removed exact path  for both top and bottom*/}
          {/* <Redirect from="/" to="dashboard/" /> */}
        </Route>
        {/* <ProtectedRoute path="/dashboard/" component={Dashboard}> */}
          {/* <Dashboard /> */}
        {/* </ProtectedRoute> */}
        <Route path="/dashboard/" component={Dashboard}>
          {/* removed exact path  for both top and bottom*/}
          {/* <Redirect from="/" to="dashboard/" /> */}
        </Route>
        <Route path="/dashboard/" component={Dashboard}>
          {/* removed exact path  for both top and bottom*/}
          {/* <Redirect from="/" to="dashboard/" /> */}
        </Route>

        <Route path="*">
          <Redirect from="/" to="/login" component={LogIn} />
        </Route>
      </Switch>
    </Router>
      </div>
    )
  }
}

export default App
