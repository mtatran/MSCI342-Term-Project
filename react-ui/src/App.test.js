import React from "react";
import { create } from "react-test-renderer";

import App from './App';

describe('App', () => {
  test('snapshot renders', () => {
    const component = create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
