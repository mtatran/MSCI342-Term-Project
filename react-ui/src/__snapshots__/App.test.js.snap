// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <header
    className="App-header-blue"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
  </header>
  <div
    className="Form-container"
    style={
      Object {
        "boxSizing": "border-box",
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 1540,
        "paddingLeft": 15,
        "paddingRight": 15,
        "position": "relative",
      }
    }
  >
    <div>
      <p
        className="Title-Style"
      >
        Log In.
      </p>
    </div>
    <div
      style={
        Object {
          "alignItems": "normal",
          "display": "flex",
          "flexGrow": 0,
          "flexShrink": 0,
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
          "marginLeft": -15,
          "marginRight": -15,
        }
      }
    >
      <div
        align="center"
        style={
          Object {
            "boxSizing": "border-box",
            "flexBasis": "100%",
            "flexGrow": 0,
            "flexShrink": 0,
            "left": "auto",
            "marginLeft": "0%",
            "maxWidth": "100%",
            "minHeight": 1,
            "paddingLeft": 15,
            "paddingRight": 15,
            "position": "relative",
            "right": "auto",
            "width": "100%",
          }
        }
      >
        <form
          onSubmit={[Function]}
        >
          <label>
            <input
              className="InputFields"
              name="email"
              onChange={[Function]}
              placeholder="email"
              required={true}
              type="text"
            />
          </label>
          <br />
          <label>
            <input
              className="InputFields"
              name="password"
              onChange={[Function]}
              placeholder="password"
              required={true}
              type="password"
            />
          </label>
          <br />
          <input
            className="SubmitButton"
            type="submit"
            value="Log In!"
          />
        </form>
        <br />
        <a
          href="/signup"
          onClick={[Function]}
        >
          Don't have an account?
        </a>
      </div>
    </div>
    <span
      style={
        Object {
          "clear": "both",
          "display": "table",
        }
      }
    />
  </div>
</div>
`;
